src/Aspire.Hosting/ApplicationModel/ResourceExtensions.cs

```diff
@@ -293,45 +288,81 @@ public static async ValueTask ProcessArgumentValuesAsync(
+                var resolvedValue = await resource.ResolveValueAsync(executionContext, logger, a, null, cancellationToken).ConfigureAwait(false);
 
-                    if (resolvedValue?.Value != null)
-                    {
-                        processValue(a, resolvedValue.Value, null, resolvedValue.IsSensitive);
-                    }
-                }
-                catch (Exception ex)
+                if (resolvedValue?.Value != null)
                 {
-                    processValue(a, a.ToString(), ex, false);
+                    processValue(a, resolvedValue.Value, null, resolvedValue.IsSensitive);
                 }
             }
+            catch (Exception ex)
+            {
+                processValue(a, a.ToString(), ex, false);
+            }
         }
     }
 
     /// <summary>
-    /// Processes environment variable values for the specified resource within the given execution context.
+    /// Processes argument values for the specified resource in the given execution context.
     /// </summary>
-    /// <param name="resource">The resource from which the environment variables are retrieved and processed.</param>
-    /// <param name="executionContext">The execution context to be used for processing the environment variables.</param>
-    /// <param name="processValue">An action delegate invoked for each environment variable, providing the key, the unprocessed value, the processed value (if available), and any exception encountered during processing.</param>
-    /// <param name="logger">The logger used to log any information or errors during the environment variables processing.</param>
-    /// <param name="cancellationToken">A cancellation token to observe during the asynchronous operation.</param>
-    /// <returns>A task that represents the asynchronous operation.</returns>
-    public static async ValueTask ProcessEnvironmentVariableValuesAsync(
+    /// <param name="resource">The resource containing the argument values to process.</param>
+    /// <param name="executionContext">The execution context used during the processing of argument values.</param>
+    /// <param name="processValue">
+    /// A callback invoked for each argument value. This action provides the unprocessed value, processed string representation,
+    /// an exception if one occurs, and a boolean indicating the success of processing.
+    /// </param>
+    /// <param name="logger">The logger used for logging information or errors during the argument processing.</param>
+    /// <param name="cancellationToken">A token for cancelling the operation, if needed.</param>
+    /// <returns>A task representing the asynchronous operation.</returns>
+    public static async ValueTask ProcessArgumentValuesAsync(
+        this IResource resource,
+        DistributedApplicationExecutionContext executionContext,
+        // (unprocessed, processed, exception, isSensitive)
+        Action<object?, string?, Exception?, bool> processValue,
```

Comment:
Maybe we should pass a context here to solve tis overload problem.

Reply:
As long as the logic is shared that's fine. The main purpose was to avoid multiple implementations that had diverged from what tests were using.

Reply:
I'm going to update this to take advantage of a feature I added in DCP around simplifying public certificate config for both exes and containers and give a single unified method that'll hopefully make it easier to integrate this with deploy in the future.